File: app/src/main/java/com/mycompany/myapp/monitoring/CrashlyticsCrashReporter.java
package com.mycompany.myapp.monitoring;

import android.content.Context;

import com.crashlytics.android.Crashlytics;
import com.mycompany.myapp.app.ForApplicationScope;

import javax.inject.Inject;

public class CrashlyticsCrashReporter implements CrashReporter {
    @Inject
    @ForApplicationScope
    Context context;

    @Override
    public void startCrashReporter() {
        Crashlytics.start(context);
    }

    @Override
    public void logMessage(String message) {
        Crashlytics.log(message);
    }
}

File: app/src/main/java/com/mycompany/myapp/monitoring/CrashlyticsCrashReporter.kt
package com.mycompany.myapp.monitoring

import com.crashlytics.android.Crashlytics

class CrashlyticsCrashReporter : CrashReporter {
    override fun logMessage(message: String) = Crashlytics.log(message)

    override fun logException(message: String, ex: Exception) {
        Crashlytics.log(message)
        Crashlytics.logException(ex)
    }
}
